### ISSUE TITLE
ex_casA_image_cat example doesn't return any results

### ISSUE THREAD

**Issue by cbanek** [astropy/pyvo#170]
After using SkyCoord, I noticed that the ex_casA_image_cat.py example still doesn't return anything.

Everything runs, but it looks like it finds no archives based on the parameters it is given:

```
pyvo.regsearch(servicetype='image', waveband='xray')
```

Returns an empty table.  I'm guessing it did return things at some point.  If I get rid of the waveband parameter, it returns a whole bunch.  I'm not sure how the example should change, but it's somewhat bad form to have an example that doesn't return results.

**Comment by cbanek** [comment]
Also, the other example works fine and makes fits images!

**Comment by msdemlei** [comment]
Hi Christine,

On Wed, Aug 28, 2019 at 02:15:54PM -0700, Christine Banek wrote:
> After using SkyCoord, I noticed that the ex_casA_image_cat.py example still doesn't return anything.
> 
> Everything runs, but it looks like it finds no archives based on the parameters it is given:
> 
> ```
> pyvo.regsearch(servicetype='image', waveband='xray')
> ```
> 
> Returns an empty table.  I'm guessing it did return things at some

That's because "xray" was a magic term used by the proprietary STScI
registry (or so I guess).  As per VODataService (the vs:Waveband type, 
http://docs.g-vo.org/schemadoc/schemas/VODataService-v1_1_xsd/simpleTypes/Waveband.html)
it's "X-ray", and by RegTAP policy of lowercasing all strings not
intended for display, it becomes x-ray in the RegTAP interface.  I
wonder if there shouldn't be things like registry.X_RAY and
registry.OPTICAL containing these magic strings...

Anyway, there's more breakage in this script.  As I've said in 
https://github.com/astropy/pyvo/issues/165#issuecomment-525192758,
I've made a working version of it; github's mail interface swallows
attachments, though, so the fixed script never made it to the issue.  

If you insist, I can turn it into a pull request, but since it's
short and essentially replaces the entire file, I'll put it here:

```
#! /usr/bin/env python
import pyvo as vo
from astropy.coordinates import SkyCoord

import warnings
warnings.filterwarnings('ignore', module="astropy.io.votable.*")

# find archives with x-ray images
archives = vo.regsearch(servicetype='image', waveband='x-ray')
# position of my favorite source
pos = SkyCoord.from_name('Cas A')

# find images and list in a file
with open('cas-a.csv', 'w') as csv:
    print("Archive short name,Archive title,Image"
          "title,RA,Dec,format,URL", file=csv)
    for arch in archives:
        print("searching {}...".format(arch.short_name))
        try:
            matches = arch.search(pos=pos, size=0.25)
        except vo.DALAccessError as ex:
            print("Trouble accessing {} archive ({})".format(
                  arch.short_name, str(ex)))
            continue
        print("...found {} images.".format(len(matches)))
        for image in matches:
            print(','.join(
             (arch.short_name, arch.res_title, image.title,
              str(image.pos), image.format,
              image.getdataurl())), file=csv)
```

        -- Markus

**Comment by cbanek** [comment]
Excellent, thanks Markus!

I just tried to run it and I get one error, but a lot of results, so this is much better.

Here's the one error I got, which seems to be a server side thing?

Trouble accessing Chandra archive (ValueError: None:119:4: ValueError: invalid literal for int() with base 10: '1024.0' (in row 0, col 'SIA_naxis'))

I'll put this up as a PR

**Comment by msdemlei** [comment]
On Thu, Aug 29, 2019 at 12:49:41PM -0700, Christine Banek wrote:
> I just tried to run it and I get one error, but a lot of results,
> so this is much better.
> 
> Here's the one error I got, which seems to be a server side thing?
> 
> Trouble accessing Chandra archive (ValueError: None:119:4:
> ValueError: invalid literal for int() with base 10: '1024.0' (in
> row 0, col 'SIA_naxis'))

Yes, that's their fault.  They're giving "1024.0 1025.0" as a literal
for datatype="int" arraysize="*"; this is perhaps not the ideal error
message (I'd wish that to be: "Row 1, column SIA_naxis: Invalid
literal for int[*]: '1024.0 1025.0'", but then I'm afraid I'm too
lazy at this point to fix it in astropy's VOTable code, and even to
make a proper bug report).

I'll contact the operators to fix it.  For the record, the RegTAP
query to figure out who to complain to is

```
SELECT *
FROM rr.res_role
WHERE
  ivoid='ivo://nasa.heasarc/chanmaster'
  AND base_role='contact'
```

I'm tempted to add a get_contact() method to RegistryResult that
encapsulates this.  

While I'm rambling anyway: I promise to put some solid work into the
registry interface late this year; for instance, currently, byte
strings are handed down for whatever fields are returned from TAP as
char[]; I'm not convinced that's actually a good thing to do in
astropy.votable, but it's definitely wrong in the return value of
regsearch.

In other words: Feel free to poke me if there are no PRs by Dec 1.
